indico/web/forms/fields.py

```diff
@@ -261,7 +273,9 @@ def _convert_principal(self, principal):
         return principal.as_principal if self.serializable else principal
 
     def process_formdata(self, valuelist):
-        if valuelist:
+        if _preprocessed_formdata(valuelist):
+            self.data = valuelist[0]
```

Comment:
same thing here (or we could assume that those complex fields are only used using actual forms and thus don't need any handling for preprocessed data)